### DISCUSSION TITLE
[PSD13] Slide 2 - Esercizio 2

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#188] [Esercizi (A-L)]
2)  Realizzare la macchina a stati dell’Esercizio 3.26 (#185 ) usando 1 FF per stato (one-hot encoding).

**Comment by Elia-Belli** [comment]
Il ragionamento dietro il diagramma di transizione è qui: https://github.com/sapienzastudentsnetwork/psd2223/discussions/185#discussioncomment-9052124.

Per praticità nel disegno, anzichè utilizzare i simboli o le codifiche degli input e del resto, ho indicato direttamente in notazione decimale l'input e il resto: la notazione è questa $Input/ Dispense, Resto$ .

![Soda](https://user-images.githubusercontent.com/107944476/215081975-b31e9f11-03a8-45b7-988e-2b98044c3b6b.jpg)

**One-Hot Encondings**: $S0=00001;\ S1=00010;\ S2=00100;\ S3=01000;\ S4=10000;$

![EL](https://user-images.githubusercontent.com/107944476/215081962-d333405d-3a29-403d-b106-45e09300e3ee.png)

Ricavare le equazioni e realizzare il circuito è banale ma laborioso, lascio l'esercizio al lettore.
